dash/backend/src/modules/auth/services/oauth2/google-oauth2.service.ts

```diff
@@ -43,9 +44,15 @@ export class GoogleOauth2Service extends Oauth2AuthProvider {
 
     userProfile.authorities = [];
 
+    // Check access rights for profile
     const userAuthority = new UserAuthority();
-    userAuthority.id = 3;
-    userProfile.authorities.push(userAuthority);
+    const emailDomain = userProfile.email.split('@').pop().toLowerCase().trim();
+    if (oAuth2Config.allowedDomains.includes(emailDomain)) {
+      userAuthority.id = AuthorityId.READ_ONLY;
+      userProfile.authorities.push(userAuthority);
```

Comment:
Right now, the OAuth flow pulls information from the OAuth provider and creates a default user, then it checks to see if a user with that email already exists in the system; if that user exists, it ignores the default user and loads in the existing user (assuming the user is active, it throws an error if the user exists but is inactive), and if the user doesn't exist it creates a new user with the information from OAuth. See the redirectCallbackHandler function [here](https://github.com/m9sweeper/m9sweeper/blob/b8289647c28681680b09af5246dcb196f8b8c5a4/dash/backend/src/modules/auth/controllers/redirectable-login.controller.ts#L69)

It probably would make more sense to have that logic in the service layer, I could move it there if you want? It would probably make the code easier to follow.